main.py

```diff
@@ -0,0 +1,364 @@
+import requests
+import json
+import datetime
+import os
+from operator import itemgetter
+
+
+# GLOBALS
+PATH = os.path.dirname(os.path.realpath(__file__))   # Gets the absolute path for the running file
+RAW_PATH = PATH + "/data/{}/raw/"
+EDIT_PATH = PATH + "/data/{}/edited/"
+FILE_NAME = "Riven_data_{}_{}.json"
+TOTAL_FILE_PATH = EDIT_PATH + "TOTAL_{}.json"
+L_MONDAY = ""
+PLATFORMS = ["PC", "PS4", "XB1", "SWI"]
+URL = "https://n9e5v4d8.ssl.hwcdn.net/repos/weeklyRivens{}.json"
+
+
+def last_monday():
+    '''Gets the last monday date'''
+    today = datetime.date.today()
+    monday = today - datetime.timedelta(days=today.weekday())
+    return monday
+
+
+def get_riven_json(platform):
+    '''Gets the json for a given platform'''
+    print(f"Checking if this week json for {platform} is saved!")
+    filename = FILE_NAME.format(platform, L_MONDAY.strftime("%y-%m-%d"))
+    file_path = f"{RAW_PATH.format(platform)}{filename}"
+
+    # Checks if the file already exists
+    if not os.path.isfile(file_path):
+        print("No file found, creating it!\n-----------------")
+        r = requests.get(URL.format(platform))
+        r.raise_for_status()
+        data = r.json()
+        with open(file_path, "w") as outfile:
+            json.dump(data, outfile, indent=4)
+    else:
+        print("File already exists!\n-----------------")
+
+
+def process_data(platform):
+    '''Calls all of the processing functions for every raw file'''
+
+    # Goes over every file in the raw folder and processes the data for it
+    for filename in os.listdir(RAW_PATH.format(platform)):
+
+        print(f"\nProcessing {filename}\n")
+
+        date = filename.rstrip(".json")[-8:]
+        raw_file_path = f"{RAW_PATH.format(platform)}{filename}"
+        edit_file_path = f"{EDIT_PATH.format(platform)}{filename}"
+
+        with open(raw_file_path, "r") as raw:
+            raw_data = json.load(raw)
+            sales = sale_calc(raw_data)
+            raw_dict = create_dict(raw_data, 1)
+            del raw_data
+
+        # Checks if the raw data is in the TOTAL file and returns the total dict
+        total_dict = total_json(platform, raw_dict, date, sales)
+
+        if os.path.isfile(edit_file_path):
+            print(f"/edited/{filename} Aleady exists!")
+        else:
+            comparison(platform, raw_dict, total_dict, sales, filename)
+
+
+def sale_calc(data):
+    '''Calculates the number of sales'''
+    print("Calculating the total sales number!")
+    min_price = []
+    max_price = []
+    pop = []
+    for d in data:
+        min_price.append(d["min"])
+        max_price.append(d["max"])
+        pop.append(d["pop"])
+
+    min_pop = min(pop)
+    pop_ind = pop.index(min_pop)
+    if min_price[pop_ind] == max_price[pop_ind]:
```

Comment:
Either way is fine so I'm gonna change it.
On a side note the whole calculating of the number of sales will work only till there is a week where no riven sold a single copy, when (or if) that happens a lot of stuff will break since even though I return None I did not write anything to really handle that scenario. It's something I plan to fix but with almost 600 possible rivens to sell It's not something I expect to happen.

Reply:
Ahaha alright 😆